core/state_snapshot.go

```diff
@@ -87,3 +87,7 @@ func (s *stateSnapshot) Class(classHash *felt.Felt) (*DeclaredClass, error) {
 	}
 	return declaredClass, nil
 }
+
+func (s *stateSnapshot) CompiledClassHash(classHash *felt.Felt) (*felt.Felt, error) {
+	return s.state.CompiledClassHash(classHash)
```

Comment:
This will not work, correct? We probably need a

```go
ClassIsDeclaredAt(classHash *felt.Felt, blockNumber uint64) (bool, error)
```
in the `StateHistoryReader` interface. That may lead to a pretty costly migration, if I'm thinking of this correctly.

Reply:
Yeah, on its own, this wont suffice. But I don't think we need a migration to get it working.
Doing something similar to what `(s *stateSnapshot) Class` is doing should be enough

But we [can remove](https://github.com/NethermindEth/juno/pull/1537#discussion_r1422029257) `CompiledClassHash` all together.
 
``` golang
	declaredClass, err := s.state.Class(classHash)
	if err != nil {
		return nil, err
	}

	if s.blockNumber < declaredClass.At {
		return nil, db.ErrKeyNotFound
	}
```